### PR TITLE
WP Stories: keep state

### PR DESCRIPTION
**Pull request by mzorz**
This PR provides support for Stories serialization / state keeping, particularly used when the StoryComposerActivity is sent to the background so we don't lose users work.

To test:
See steps in https://github.com/Automattic/portkey-android/pull/435

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 1]
```diff
@@ -902,6 +902,9 @@ private void setSite(Intent data) {
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
+        if (mSelectedSite == null) {
+            initSelectedSite();
+        }
```
Comment (jd-alexander) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:907]: Hey @mzorz could you give me some more context why this change was necessary for this PR?

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:907]: oh sure thing! Forgot to mention this. When you set the `Don't keep activities` to ON in developer settings, you'll find that just sitting there on the main screen of WPAndroid, sending it to the background and then bringing it to the foreground will do strange things because the `mSelectedSite` is lost. There are safeguards for this same situation elsewhere (just do a search for `initSelectedSite()` and you'll see ;) ).
This is particularly needed if you want to create a story because our flow there is we first start the media picker for result (and at that point the main activity is gone!), as soon as you have chosen something, get back for onActivityResult in the main activity but the selected site is lost so you'll be sitting there 😅 .
